apps/full-site-editing/full-site-editing-plugin/donations/stripe-nudge.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+/**
+ * WordPress dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import { Button, Dashicon } from '@wordpress/components';
+import { Warning } from '@wordpress/block-editor';
+import { compose } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+import { withDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+
+const StripeNudge = ( { autosaveAndRedirect, stripeConnectUrl } ) => (
+	<Warning
+		actions={
+			// Use href to determine whether or not to display the Upgrade button.
+			stripeConnectUrl && [
+				<Button
+					key="connect"
+					href={ stripeConnectUrl } // Only for server-side rendering, since onClick doesn't work there.
+					onClick={ autosaveAndRedirect }
+					target="_top"
+					isDefault
+					className="premium-content-block-nudge__button stripe-nudge__button"
+				>
+					{ __( 'Connect', 'full-site-editing' ) }
+				</Button>,
+			]
+		}
+		className="donations__nudge premium-content-block-nudge"
+	>
+		<span className="donations__nudge-info premium-content-block-nudge__info">
+			<Dashicon icon="star-filled" />
+			<span className="donations__nudge-text-container premium-content-block-nudge__text-container">
+				<span className="donations__nudge-title premium-content-block-nudge__title">
+					{ __( 'Connect to Stripe to use this block on your site', 'full-site-editing' ) }
+				</span>
+				<span className="donations__nudge-message premium-content-block-nudge__message">
+					{ __(
+						'This block will be hidden from your visitors until you connect to Stripe.',
```

Comment:
I guess this will be handled in a follow-up PR, right? Just noted the saved content is still visible in the frontend regardless the Stripe connection status.

Reply:
Yes, we can handle that when dealing with the front-end 👍